### DISCUSSION TITLE
AggregatorV3Interface question

### DISCUSSION THREAD

**Discussion by dvelez3815** [Cyfrin/foundry-full-course-cu#2319] [Q&A]
How it's possible to create a instance of a interface?

How it's possible to do something like AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306).version;

1.- We know we can't not make instance of interface.
2.- Interface do not receive params in the constructor since it's not possible to create a instance, so how it's explain this line of code:

AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306).version
if we don't actuality create a new object how it works internally, what object is built in order to execute .version function? The address has the ABI reference and the ABI has the function definition but how to initialize a interface with an address make sense?  I know it works, but I want to know whyy it's possible this: AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306)

thanks.

**Comment by EngrPips** [comment]
Hello @dvelez3815, so this line `AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306)` basically creates a contract object because `interface + address = contract object` and on the contract object you can call any function that is on the interface. This basically means it will use the interface to get the function selector of the function you want to call, and under the hood, it will use the `call` method to call the function using the `function selector` it generated on the address, which we wrapped inside the interface. 

```js
//We have an address of a priceFeed contract that is deployed here `0x694AA1769357215DE4FAC081bf1f309aDC325306` 
// wrap that address with `AggregatorV3Interface` to form a contract object that we can call functions on to directly make a call to the smart contract at this address `AggregatorV3Interface` 
//Call the `version` method on the contract object to get the `version` that is currently stored in the smart contract at this address //`0x694AA1769357215DE4FAC081bf1f309aDC325306` 
AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306).version
```
Please ask questions if anything is still confusing.
